### PR TITLE
Grow 1563 Add analytics tracking to Nav Bar

### PR DESCRIPTION
**Pull request by jiahaodavid**
Address: [Grow 1563](https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=4&projectKey=GROW&modal=detail&selectedIssue=GROW-1563)

This PR is going to add tracking "Hover" action to our code. It already tracked the "Click" action. So, I set up a condition when `haver && doHoverTracking` is true, then they will do the hover tracking.

### CODE REVIEW

File: src/Components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -83,20 +83,29 @@ export const NavBar: React.FC = track(
         */}
         <NavSection display={["none", "none", "flex"]}>
           <NavSection>
-            <NavItem href="/collect">Artworks</NavItem>
-            <NavItem href="/auctions">Auctions</NavItem>
-            <NavItem href="/galleries">Galleries</NavItem>
+            <NavItem href="/collect" doHoverTracking>
```
Comment (damassi) [inline src/Components/NavBar/NavBar.tsx]: Why add the `doHoverTracking` prop if all nav items are tracked on hover?

Comment (jiahaodavid) [reply src/Components/NavBar/NavBar.tsx]: @damassi 
Yes. Because there are somethings also using `<NavItem>`. Such as, "works for you" and some small view. For "work for you", it already has a hover tracking. I think only adding `doHoverTracking` props to those items will minimus the changing of the code. What do you think?

Comment (damassi) [reply src/Components/NavBar/NavBar.tsx]: I would suggest creating a new prop `onHover` that can be passed into `NavItem`, and for  that single nav item you mention that already tracks we can place the hover tracking in that callback function. Inside of NavItem, if that `onHover` prop has been passed, guard against the default hover tracking (default = what happens `doHoverTracking` is passed, at which point we can remove the `doHoverTracking` prop completely).

Basically, if most of  the items now require hover tracking, then a prop shouldn't have  to be passed because one nav item already has it. Things can be inverted. Revise the item that already has hover tracking and make it more suitable for the many. 

Does this make sense? Down to pair if needed!

Comment (jiahaodavid) [reply src/Components/NavBar/NavBar.tsx]: @damassi 
Hi, Chris. I made some updates. Take a look when you have time.

File: src/Components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -116,6 +116,7 @@ export const NavBar: React.FC = track(
             {isLoggedIn && (
               <>
                 <NavItem
+                  onHover={false}
```
Comment (damassi) [inline src/Components/NavBar/NavBar.tsx]: One should be able to exclude this prop since `false` and `undefined` (missing the prop) both evaluate to false in the end.

File: src/Components/NavBar/NavItem.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ interface NavItemProps extends BoxProps {
   className?: string
   href?: string
   onClick?: () => void
+  onHover?: boolean
```
Comment (damassi) [inline src/Components/NavBar/NavItem.tsx]: Props with the `onSomething` form are always callback functions since the `onSomeEvent` form implies some action is happening, like a hover or a mouse click (etc).

File: src/Components/NavBar/NavItem.tsx [hunk 4]
```diff
@@ -27,6 +28,7 @@ export const NavItem: React.FC<NavItemProps> = ({
   display = "block",
   href,
   onClick,
+  onHover = true,
```
Comment (damassi) [inline src/Components/NavBar/NavItem.tsx]: This isn't needed

File: src/Components/NavBar/NavItem.tsx [hunk 5]
```diff
@@ -94,6 +104,8 @@ export const NavItem: React.FC<NavItemProps> = ({
         </Sans>
       </Link>
 
+      {hover && onHover ? trackHover() : null}
```
Comment (damassi) [inline src/Components/NavBar/NavItem.tsx]: This can [be an effect](https://reactjs.org/docs/hooks-effect.html), and we can call the tracking function when `hover` is true.

```tsx
useEffect(() => {
  if (hover) {
    trackHover()
  }
}, [hover])
```

File: src/Components/NavBar/NavBar.tsx [hunk 6]
```diff
@@ -147,6 +147,12 @@ export const NavBar: React.FC = track(
                 </NavItem>
                 <NavItem Menu={UserMenu}>
                   {({ hover }) => {
+                    if (hover) {
+                      trackEvent({
+                        action_type: AnalyticsSchema.ActionType.Hover,
+                        subject: "User",
+                      })
+                    }
```
Comment (damassi) [inline src/Components/NavBar/NavBar.tsx]: 👍

File: src/Components/NavBar/NavItem.tsx [hunk 7]
```diff
@@ -59,6 +61,14 @@ export const NavItem: React.FC<NavItemProps> = ({
     }
   }
 
+  const trackHover = () => {
+    if (isString(children))
```
Comment (damassi) [inline src/Components/NavBar/NavItem.tsx]: Can we add a comment above this line mentioning that `children` refers text that's passed into `<NavItem>`.

File: src/Components/NavBar/NavItem.tsx (hunk 5 above)
Comment (damassi) [reply src/Components/NavBar/NavItem.tsx]: Another thought is functions shouldn't be called like this inside of JSX, unless the function returns more JSX. Why call in here instead of around line 72?

**Comment by jiahaodavid** [comment]
@damassi 
I also made some updates. Thanks for the feedback. 
https://github.com/artsy/reaction/blob/eadc3b68e36998aa5f325ba38cb0ed97b6b326c6/src/Components/NavBar/NavItem.tsx#L93-L95
I have a question. It is not related to this PR, but just for curiosity.
Why do you use `onClick && onClick()` here?

**Comment by damassi** [comment]
@jiahaodavid - this is shorthand for 

```jsx
if (onClick) {
  onClick()
}
```
"if x is true, then y" -- same kind of thing as 
```jsx
const isNiceDay = sunny && checkIfRaining()
```

However, I don't recommend (and typically never write) shorthand if statements and not sure why that is there! They can be buggy, sacrafice readability for "clever" shorthand, and also come with some edge-cases. Always use an `if`!

**Comment by artsyit** [comment]
:rocket: PR was released in v21.17.2 :rocket:
